webapp/src/main/java/com/box/l10n/mojito/smartling/SmartlingClient.java

```diff
@@ -0,0 +1,199 @@
+package com.box.l10n.mojito.smartling;
+
+import com.box.l10n.mojito.smartling.request.Errors;
+import com.box.l10n.mojito.smartling.response.AuthenticationResponse;
+import com.box.l10n.mojito.smartling.response.SourceStringsResponse;
+import com.ibm.icu.text.MessageFormat;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriTemplate;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.*;
+
+/**
+ * Smartling client based on the Web API: https://github.com/Smartling/api-sdk-java
+ */
+public class SmartlingClient {
+
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(SmartlingClient.class);
+
+    static final String DEFAULT_BASE_HOST = "https://api.smartling.com/";
+    static final String AUTH_API_V2_AUTHENTICATE = DEFAULT_BASE_HOST + "auth-api/v2/authenticate";
+    static final String AUTH_API_V2_REFRESH = DEFAULT_BASE_HOST + "auth-api/v2/authenticate/refresh";
+    static final String API_PULL_SOURCE_STRINGS = DEFAULT_BASE_HOST + "strings-api/v2/projects/{projectId}/source-strings?fileUri={fileUri}&offset={offset}";
+    static final String API_PUSH_NEW_CONTEXT = DEFAULT_BASE_HOST + "context-api/v2/projects/{projectId}/contexts";
+    static final String API_BIND_CONTEXT_TO_HASHCODE = DEFAULT_BASE_HOST + "context-api/v2/projects/{projecId}/bindings";
+
+    RestTemplate restTemplate = new RestTemplate();
+
+    String userIdentifier;
+    String userSecret;
+    String authToken = null;
+    String refreshToken = null;
+    Instant lastAuthentication = null;
+    Instant lastRefresh = null;
+    Instant nextAuthentication = null;
+    Instant nextRefresh = null;
+    // Refresh tokens are valid for up to 1 hour
+    Integer refreshDurationMinutes = 50;
+    // Authentication tokens can be refreshed for up to 24 hours
+    Integer authenticationDurationHours = 23;
+
+    public SmartlingClient(String userIdentifier, String userSecret) {
+        this.userIdentifier = userIdentifier;
+        this.userSecret = userSecret;
+    }
+
+    public enum RequestType {
+        AUTHENTICATION,
+        REFRESH,
+        SOURCE_STRINGS,
+        WITH_TOKEN
+    }
+
+    Map<String, Object> getPayloadMap(RequestType requestType) {
```

Comment:
probably better to have 2 independent functions and not pass the `requestType` in that case